java/arcs/core/util/Json.kt

```diff
@@ -76,7 +76,33 @@ sealed class JsonValue<T>() {
 
     /** Used to represent a Json string */
     data class JsonString(override val value: String) : JsonValue<String>() {
-        override fun toString() = "\"${value.replace("\"", "\\\"")}\""
+        private val escapes = mapOf(
+            "\\" to "\\\\",
+            "\"" to "\\\"",
+            "\b" to "\\b",
+            12.toChar().toString() to "\\f",
+            "\n" to "\\n",
+            "\r" to "\\r",
+            "\t" to "\\t"
+        )
+
+        private fun escape(string: String): String {
+            var rtn = string
+            escapes.forEach { old, new ->
```

Comment:
this can be written as
```
val rtn = escapes.entries.fold(string) { old, new -> old.replace(new) }
```
or if you want to one-line it

```
fun escape(string: String) = escapes.entries.fold(string) { old, new -> 
    old.replace(new) 
}.replace(Regex("[^\u0000-\u007F]")) { match ->
   match.value.toCharArray.fold("") { unicode, char -> unicode + "\\u${char.toInt()}" }
}
```